hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -288,159 +303,196 @@ public void setFilesystemProperties(final Hashtable<String, String> properties)
+      } catch (CharacterCodingException ex) {
+        throw new InvalidAbfsRestOperationException(ex);
+      }
+      final AbfsRestOperation op = client.setPathProperties(AbfsHttpConstants.FORWARD_SLASH + getRelativePath(path), commaSeparatedProperties);
+      perfInfo.registerResult(op.getResult()).registerSuccess(true);
     }
-    client.setPathProperties(AbfsHttpConstants.FORWARD_SLASH + getRelativePath(path), commaSeparatedProperties);
   }
 
   public void createFilesystem() throws AzureBlobFileSystemException {
-    LOG.debug("createFilesystem for filesystem: {}",
-            client.getFileSystem());
+    try (AbfsPerfInfo perfInfo = startTracking("createFilesystem", "createFilesystem")){
+      LOG.debug("createFilesystem for filesystem: {}",
+              client.getFileSystem());
 
-    client.createFilesystem();
+      final AbfsRestOperation op = client.createFilesystem();
+      perfInfo.registerResult(op.getResult()).registerSuccess(true);
+    }
   }
 
   public void deleteFilesystem() throws AzureBlobFileSystemException {
-    LOG.debug("deleteFilesystem for filesystem: {}",
-            client.getFileSystem());
+    try (AbfsPerfInfo perfInfo = startTracking("deleteFilesystem", "deleteFilesystem")) {
+      LOG.debug("deleteFilesystem for filesystem: {}",
+              client.getFileSystem());
 
-    client.deleteFilesystem();
+      final AbfsRestOperation op = client.deleteFilesystem();
+      perfInfo.registerResult(op.getResult()).registerSuccess(true);
+    }
   }
 
   public OutputStream createFile(final Path path, final boolean overwrite, final FsPermission permission,
                                  final FsPermission umask) throws AzureBlobFileSystemException {
-    boolean isNamespaceEnabled = getIsNamespaceEnabled();
-    LOG.debug("createFile filesystem: {} path: {} overwrite: {} permission: {} umask: {} isNamespaceEnabled: {}",
-            client.getFileSystem(),
-            path,
-            overwrite,
-            permission.toString(),
-            umask.toString(),
-            isNamespaceEnabled);
-
-    client.createPath(AbfsHttpConstants.FORWARD_SLASH + getRelativePath(path), true, overwrite,
-        isNamespaceEnabled ? getOctalNotation(permission) : null,
-        isNamespaceEnabled ? getOctalNotation(umask) : null);
-
-    return new AbfsOutputStream(
-        client,
-        AbfsHttpConstants.FORWARD_SLASH + getRelativePath(path),
-        0,
-        abfsConfiguration.getWriteBufferSize(),
-        abfsConfiguration.isFlushEnabled(),
-        abfsConfiguration.isOutputStreamFlushDisabled());
+    try (AbfsPerfInfo perfInfo = startTracking("createFile", "createPath")) {
+      boolean isNamespaceEnabled = getIsNamespaceEnabled();
+      LOG.debug("createFile filesystem: {} path: {} overwrite: {} permission: {} umask: {} isNamespaceEnabled: {}",
+              client.getFileSystem(),
+              path,
+              overwrite,
+              permission.toString(),
```

Comment:
Avoid the toString()

Reply:
@goiri This code is not added by me -- there is just a change in indentation here.

Reply:
Follow up JIRA to remove those.

Reply:
@goiri Follow-up JIRA here: https://issues.apache.org/jira/browse/HADOOP-16682